src/rgw/rgw_rest_pubsub.cc

```diff
@@ -91,13 +92,18 @@ int verify_topic_owner_or_policy(req_state* const s,
   }
   // no policy set.
   if (topic.policy_text.empty()) {
-    // if mandatory_topic_permissions is true, then validate all users for
-    // permission.
-    if (s->cct->_conf->mandatory_topic_permissions) {
-      return -EACCES;
-    } else {
+    // if rgw_topic_require_publish_policy is "false" dont validate "publish" policies
+    if (op == rgw::IAM::snsPublish && !s->cct->_conf->rgw_topic_require_publish_policy) {
+      return 0;
+    }
+    if (topic.user.empty()) {
+      // if we don't know the original user and there is no policy
+      // we will not reject the request.
+      // this is for compatibility with versions that did not store the user in the topic
       return 0;
     }
+    s->err.message = "Topic was already created by another user.";
```

Comment:
a nit: This error message should be there only for `CreateTopic`, so it would make sense to have this error while failure in CreateTopic.
This function is common wrapper called for all sns calls. for example if some other user calls the GetTopic with no policy it will get error message `Topic was already created by another user`,  which sounds in correct ?

Reply:
you are right, so maybe just `Topic was created by another user` ?